brain/detail/compartmentReport.h

```diff
@@ -72,27 +72,36 @@ struct CompartmentReportView
 void CompartmentReportView::_initIndices()
 {
     size_t indicesCount = 0;
-    size_t index = 0;
-    auto gidCount = report->getGIDs().size();
+    const auto& gids = report->getGIDs();
 
-    while (gidCount--)
-        indicesCount += report->getOffsets()[index++].size();
+    auto gidCount = gids.size();
+    std::vector<size_t> indexPositions(gidCount);
+    for (size_t i = 0; i < gids.size(); ++i)
+    {
+        indexPositions[i] = indicesCount;
+        indicesCount += report->getOffsets()[i].size();
+    }
 
-    indices.reserve(indicesCount);
-    index = 0;
+    indices.resize(indicesCount);
 
-    for (auto gid : report->getGIDs())
+    std::vector<uint32_t> gidList(report->getGIDs().begin(),
+                                  report->getGIDs().end());
+// With 8 threads there's some speed up, but very small. We stick with 6
+// to avoid using more than 1 socket.
+#pragma omp parallel for num_threads(6)
```

Comment:
According to msdn, num_threads is the maximum number of threads that will execute the parallel region, unless dynamic adjustment is disabled, in which case the result of overcommiting is implementation defined (it's even allowed to abort :(). This clause also overrides the effect of OMP_NUM_THREADS
Something I've observed for sure, is that due to the NUMA architecture, going to another socket becomes slower (even more than x2) and unpredictable.

I didn't use VTune because I did it remotely and I haven't spend time to learn how to use the command line tools, but now you're an "expert", we can look into it tomorrow together if you want. The only profiling I was using was measuring the time of the whole operation and "sampling" with the debugger.
Without threading it takes x3 times more.